Some other NoSuchMethod, NPEs fixed

Rafael2616 opened the pull request:
## Description

<!-- Please include a summary of the change. Please also include relevant motivation and context. -->

this fixes this issue also, when clciking in the widtget without holding it

before:
https://github.com/user-attachments/assets/a4f1bb16-5841-42e8-9dd8-07f7125c064a

after:
nothing will happens if only click in the widget

Fixes #(issue) <!-- optional -->

## Type of change
<!-- Replace :x: with :white_check_mark: to "check" the specified bullet -->

:x: General change (non-breaking change that doesn't fit the below categories like copyediting)
:white_check_mark: Bug fix (non-breaking change which fixes an issue)
:x: New feature (non-breaking change which adds functionality)
:x: Breaking change (fix or feature that would cause existing functionality to not work as expected)

Rafael2616 committed:
fixed java.lang.NoSuchMethodError: No interface method toList()

´´´
Ljava/util/List; in class Ljava/util/stream/Stream; or its super classes (declaration of 'java.util.stream.Stream' appears in /system/framework/core-oj.jar)
                                                                                                                            	at com.android.launcher3.model.WidgetsModel.getFilteredWidgetsListForPicker(WidgetsModel.java:99)
                                                                                                                            	at com.android.launcher3.model.WidgetsModel.getWidgetsListForPicker(WidgetsModel.java:122)
                                                                                                                            	at com.android.launcher3.model.BaseLauncherBinder.bindWidgets(BaseLauncherBinder.java:204)
                                                                                                                            	at com.android.launcher3.model.LoaderTask.run(LoaderTask.java:338)
                                                                                                                            	at android.os.Handler.handleCallback(Handler.java:873)
                                                                                                                            	at android.os.Handler.dispatchMessage(Handler.java:99)
                                                                                                                            	at android.os.Looper.loop(Looper.java:193)
                                                                                                                            	at android.os.HandlerThread.run(HandlerThread.java:65)

                                                                                                                            	´´´

Rafael2616 committed:
Merge branch 'LawnchairLauncher:15-dev' into 15-dev

Rafael2616 committed:
fix crashes when trying to customize from folders

Rafael2616 committed:
Merge remote-tracking branch 'origin/15-dev' into 15-dev

Rafael2616 committed:
Merge branch 'LawnchairLauncher:15-dev' into 15-dev

Rafael2616 committed:
fix NoClassDefFound attempting to create new folder in lower apis

Rafael2616 committed:
fix java.lang.NoSuchMethodError: No static method beginAsyncSection(Ljava/lang/String;I)V in class Landroid/os/Trace; or its super classes (declaration of 'android.os.Trace' appears in /system/framework/framework.jar!classes2.dex) when adding widgets below Q

                                                                                                        	at com.android.launcher3.widget.LauncherAppWidgetHostView.setAppWidget(LauncherAppWidgetHostView.java:126)
                                                                                                        	at android.appwidget.AppWidgetHost.createView(AppWidgetHost.java:382)
                                                                                                        	at com.android.launcher3.widget.LauncherWidgetHolder.createViewInternal(LauncherWidgetHolder.java:447)
                                                                                                        	at com.android.launcher3.widget.LauncherWidgetHolder.createView(LauncherWi

Rafael2616 committed:
fix java.lang.NoSuchMethodError: No static method beginAsyncSection(Ljava/lang/String;I)V in class Landroid/os/Trace; or its super classes (declaration of 'android.os.Trace' appears in /system/framework/framework.jar!classes2.dex) when adding widgets below Q

                                                                                                        	at com.android.launcher3.widget.LauncherAppWidgetHostView.setAppWidget(LauncherAppWidgetHostView.java:126)
                                                                                                        	at android.appwidget.AppWidgetHost.createView(AppWidgetHost.java:382)
                                                                                                        	at com.android.launcher3.widget.LauncherWidgetHolder.createViewInternal(LauncherWidgetHolder.java:447)
                                                                                                        	at com.android.launcher3.widget.LauncherWidgetHolder.createView(LauncherWi

Rafael2616 committed:
fix                                                                                                     java.lang.NoSuchMethodError: No virtual method setSystemGestureExclusionRects(Ljava/util/List;)V in class Lcom/android/launcher3/AppWidgetResizeFrame; or its super classes (declaration of 'com.android.launcher3.AppWidgetResizeFrame' appears in /data/app/app.lawnchair.play.debug-ewPju7gaDxUbh-ipLPLIoA==/base.apk!classes6.dex)

Rafael2616 committed:
Merge remote-tracking branch 'origin/15-dev' into 15-dev

Rafael2616 committed:
fix more NoSuchMethodError

Rafael2616 committed:
fix NPE´s

Rafael2616 committed:
fix: disable showing addButton widget as it not make sense i think

MrSluffy commented on src/com/android/launcher3/widget/BaseWidgetSheet.java:
```diff
@@ -163,12 +163,12 @@ public final void onClick(View v) {
-                    wc.showAddButton((view) -> mActivityContext.getItemOnClickListener()
-                            .onClick(wc));
-                }
+//                if (mActivityContext instanceof Launcher) {
```
why are we commenting this? is there no work arround?

Rafael2616 replied:
this is the code that makes that artifact appear when clicking on widgets without pressing them, I actually don't know why this is useful, so I just commented it so I don't lose sight of it

The behavior is shown in the video

validcube replied:
> this is the code that makes that artifact appear when clicking on widgets without pressing them, I actually don't know why this is useful, so I just commented it so I don't lose sight of it
> 
> The behavior is shown in the video

It's likely part of Android 15 feature here

![image](https://github.com/user-attachments/assets/0f43a7a3-3ce1-488d-b3f9-2b24f2a4fc12)

Rafael2616 replied:
Thanks for the info, I'll update the PR taking this into consideration

Rafael2616 replied:
Do you know if this is only available in Android 15, or is it also present in Android 14 or other versions?

Rafael2616 replied:
@MrSluffy should i isolate this by Android Version ?? as i thinkt this is not expected to work before a15?

validcube replied:
> Do you know if this is only available in Android 15, or is it also present in Android 14 or other versions?

Hello, sorry for the slow response. This feature or "hint" is only available in Android 15 (as of Beta 2)

MrSluffy commented:
and the widget shortcut icons seems to be broken, I might need to check this

Rafael2616 commented:
> and the widget shortcut icons seems to be broken, I might need to check this

This is not related to this PR, this issue was already present

Rafael2616 committed:
fix  java.lang.NoSuchMethodError: No virtual method getInstallerUid()I in class Landroid/content/pm/PackageInstaller$SessionInfo; or its super classes (declaration of 'android.content.pm.PackageInstaller$SessionInfo' appears in /system/framework/framework.jar)
                                                                                                        	at com.android.launcher3.pm.InstallSessionHelper.getUserHandle(InstallSessionHelper.java:265)
                                                                                                        	at com.android.launcher3.pm.InstallSessionHelper.verify(InstallSessionHelper.java:164)

Rafael2616 committed:
fix  java.lang.NoSuchMethodError: No virtual method getInstallerUid()I in class Landroid/content/pm/PackageInstaller$SessionInfo; or its super classes (declaration of 'android.content.pm.PackageInstaller$SessionInfo' appears in /system/framework/framework.jar)
                                                                                                        	at com.android.launcher3.pm.InstallSessionHelper.getUserHandle(InstallSessionHelper.java:265)
                                                                                                        	at com.android.launcher3.pm.InstallSessionHelper.verify(InstallSessionHelper.java:164)

MrSluffy commented on src/com/android/launcher3/pm/InstallSessionHelper.java:
```diff
@@ -262,6 +262,6 @@ public InstallSessionTracker registerInstallTracker(
 
     public static UserHandle getUserHandle(@NonNull final SessionInfo info) {
-        return info.getUser();
+        return Utilities.ATLEAST_Q ? info.getUser() : UserHandle.getUserHandleForUid(info.getSessionId());
```
Please use Process.myUserHandle() or UserHandle.getUserHandleForUid(info.userId) instead

Rafael2616 replied:
Right

Rafael2616 committed:
Update InstallSessionHelper

Rafael2616 committed:
Merge branch 'LawnchairLauncher:15-dev' into 15-dev

Rafael2616 committed:
Merge remote-tracking branch 'origin/15-dev' into 15-dev

Rafael2616 committed:
revert changes in BaseWidgetSheet

Rafael2616 commented:
i just revert the changes in `BaseWidgetSheet` 
so that only changes related to locking corrections remain in this PR @MrSluffy can you check this

MrSluffy commented on src/com/android/launcher3/compat/AccessibilityManagerCompat.java:
```diff
@@ -110,6 +112,9 @@ private static AccessibilityManager getAccessibilityManagerForTest(Context conte
     public static int getRecommendedTimeoutMillis(Context context, int originalTimeout, int flags) {
-        return getManager(context).getRecommendedTimeoutMillis(originalTimeout, flags);
+        if (ATLEAST_Q) {
+            return getManager(context).getRecommendedTimeoutMillis(originalTimeout, flags);
```
use ternary instead return AT LEAST Q ? getManager(context).getRecommendedTimeoutMillis(originalTimeout, flags) : originalTimeout

Rafael2616 replied:
done

Rafael2616 committed:
apply changes from code review
